skimage/metrics/overlap.py

```diff
@@ -0,0 +1,251 @@
+
+        raise NotImplementedError("Volume is only defined for 3D.")
+
+    @property
+    def integral(self):
+        """
+        Return the integral of the shape along all dimensions of the shape.
+
+        For 2D/3D shapes, the integral corresponds respectively to the area/volume.
+        """
+        return np.prod(self.dimensions)
+
+    @property
+    def dimensions(self):
+        """Return the dimensions of the BoundingBox as an array.
+
+        Examples
+        --------
+        >>> r = BoundingBox((1, 1), bottom_right=(2, 3))
+        >>> r.dimensions
+        array([1, 2])
+        """
+        return self.bottom_right - self.top_left
+
+
+def disjoint(bbox1, bbox2):
+    """Check whether two BoundingBoxes are disjoint.
+
+    Adapted from post from Aman Gupta [1]_.
+
+    Parameters
+    ----------
+    bbox1, bbox2 : BoundingBox
+        Input BoundingBoxes.
+
+    Returns
+    -------
+    disjoint : bool
+        True if the BoundingBoxes don't share any points.
+
+    References
+    ----------
+    .. [1] https://www.geeksforgeeks.org/find-two-rectangles-overlap/
+    """
+    disjoint = (
+            np.any(bbox1.bottom_right < bbox2.top_left)
+            or np.any(bbox2.bottom_right < bbox1.top_left)
+            )
+    return disjoint
+
+
+def intersect(bbox1, bbox2):
+    """
+    Return the BoundingBox corresponding to the intersection of 2 BoundingBoxes.
+
+    Return None if the two BoundingBoxes do not overlap.
+    The intersection of 2 BoundingBoxes can yield a 0-area BoundingBox
+    when the intersection of the BoundingBoxes is a corner or an edge.
+
+    Parameters
+    ----------
+    bbox1, bbox2 : BoundingBox
+        Input BoundingBoxes.
+
+    Returns
+    -------
+    intersect : BoundingBox
+        The BoundingBox produced by the intersection of ``bbox1`` and``bbox2``
+        or None if the BoundingBoxes are not interesecting (ie are disjoint).
+
+    Examples
+    --------
+    >>> r0 = BoundingBox((0, 0), bottom_right=(2, 3))
+    >>> r1 = BoundingBox((1, 2), bottom_right=(4, 4))
+    >>> intersect(r0, r1)
+    BoundingBox((1, 2), bottom_right=(2, 3))
+
+    >>> r2 = BoundingBox((10, 10), dimensions=(3, 3))
+    >>> if intersect(r1, r2) is None:
+    ...     print('r1 and r2 are not intersecting')
+    r1 and r2 are not intersecting
+    """
+
+    if bbox1 == bbox2:  # return a copy
+        return BoundingBox(bbox1.top_left,
+                           bottom_right=bbox1.bottom_right)
+
+    if disjoint(bbox1, bbox2):
+        return None
+
+    new_top_left = np.maximum(bbox1.top_left, bbox2.top_left)
+    new_bottom_right = np.minimum(
+            bbox1.bottom_right, bbox2.bottom_right
+            )
+    return BoundingBox(new_top_left,
+                       bottom_right=new_bottom_right)
+
+
+def intersection_over_union(bbox1, bbox2):
+    """
+    Ratio intersection over union for a pair of BoundingBoxes.
+
+    The intersection over union (IoU) ranges between 0 (no overlap) and 1 (full
```

Comment:
```suggestion
    Intersection over union (IoU) is a metric for the overlap between two bounding boxes.
    
    The IoU ranges between 0 (no overlap) and 1 (full
```

Reply:
May also be worth mentioning the Jacard index, and showing the formula from https://en.wikipedia.org/wiki/Jaccard_index